libs/community/langchain_community/tools/google_serper/tool.py

```diff
@@ -58,7 +58,7 @@ def _run(
         run_manager: Optional[CallbackManagerForToolRun] = None,
     ) -> str:
         """Use the tool."""
-        return str(self.api_wrapper.results(query))
+        return self.api_wrapper.results(query)
```

Comment:
@dopc this is a breaking change. The change itself is also not correct given that the tool claims to return `str` (see function signature). If you want to use the tool to return things that are not strings, you can use the underlying API wrapper.

@baskaryan tagging you in case there's any general refactor that can be done w/ the new `artifact` attribute in tool message

Reply:
@eyurtsev we can start returning both the raw results and the string as a tuple, with response_format=content_and_artifact. here's an example https://github.com/langchain-ai/langchain/pull/24376